### PR TITLE
Feat/400 intermediate steps + sbt assembly + spark-submit

### PR DESCRIPTION
**Pull request by creativeyann17**
ETL is now performing intermediate steps.

clinical-data/ (TSV) -> clinical-data-with-ids/ (parquet) -> clinical-data-etl-indexer/ (JSON)

and put all configuration in application.conf instead of being everywhere

- [x] should work with spark-submit (fix assembly + deps + un-necessary code)

example :
sbt clean assembly
export KEYCLOAK_SECRET_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
spark-submit --master local --packages org.apache.hadoop:hadoop-aws:3.2.0 --class ca.cqdg.etl.EtlApp target/scala-2.12/cqdg-etl.jar

### CODE REVIEW

**Review by adipaul1981 — APPROVED** [review]

File: src/main/scala/ca/cqdg/etl/package.scala [hunk 1]
```diff
@@ -12,6 +24,42 @@ package object etl {
...
+  val s3Region: String = s3Config.getString("region")
+  val s3ClientId: String =s3Config.getString("client-id")
+  val s3SecretKey: String =s3Config.getString("secret-key")
+
+  spark.sparkContext.setLogLevel("WARN")
+  spark.sparkContext.hadoopConfiguration.set("fs.s3a.endpoint", s3Endpoint)
+  spark.sparkContext.hadoopConfiguration.set("fs.s3a.path.style.access", "true")
+  spark.sparkContext.hadoopConfiguration.set("fs.s3a.access.key", s3ClientId)
+  spark.sparkContext.hadoopConfiguration.set("fs.s3a.secret.key", s3SecretKey)
+  sys.addShutdownHook(spark.stop())
```
Comment (jecos) [inline src/main/scala/ca/cqdg/etl/package.scala:50]: this should be set in env var

File: src/main/scala/ca/cqdg/etl/package.scala [hunk 2]
```diff
@@ -12,6 +24,42 @@ package object etl {
...
+  clientConfiguration.setSignerOverride("AWSS3V4SignerType")
+
+  val s3Client: AmazonS3 = AmazonS3ClientBuilder
+    .standard()
+    .withEndpointConfiguration(new EndpointConfiguration(s3Endpoint, s3Region))
+    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(s3ClientId, s3SecretKey)))
+    .withPathStyleAccessEnabled(true)
+    .withClientConfiguration(clientConfiguration)
+    .build()
+
```
Comment (jecos) [inline src/main/scala/ca/cqdg/etl/package.scala:62]: why do we need am s3 client in etl? spark should handle s3 connection

File: src/main/scala/ca/cqdg/etl/utils/S3Utils.scala [hunk 3]
```diff
@@ -22,17 +22,43 @@ object S3Utils {
...
+        (!hasMetadata || entry._2.exists(f => f.filename == "study_version_metadata.json")) &&
         !entry._2.exists(f => f.filename == "_SUCCESS"))  // Indicator that it has already been processed
     filesGroupedByKey
   }
 
+  def loadPreProcessedEntries(bucket: String, prefix: String, s3Client: AmazonS3): Map[String, List[String]] = {
+
+    val filesGroupedByKey = ls(bucket, prefix, s3Client) // list all S3 files
+      .groupBy(f => f.parentKey)
+      .filter(entry => entry._2.exists(f => f.filename == "_SUCCESS") // only if successfully saved by spark
```
Comment (jecos) [inline src/main/scala/ca/cqdg/etl/utils/S3Utils.scala:40]: deconstruct the tuple

File: src/main/scala/ca/cqdg/etl/package.scala (hunk 1 above)
Comment (creativeyann17) [reply src/main/scala/ca/cqdg/etl/package.scala:50]: env. vars. are all in application.conf

File: src/main/scala/ca/cqdg/etl/package.scala (hunk 2 above)
Comment (creativeyann17) [reply src/main/scala/ca/cqdg/etl/package.scala:62]: to write the custom _SUCCESS file + when looking for new files to process

**Pull request merged**